Validator: Build correct required peers set

omerfirmak opened the pull request:
Not tested yet, but hoping that it will help us eliminate some of the SCP traffic.

omerfirmak committed:
FullNode: getValidators() should default to height + 1

Activate validators are defined as the validator set for the next
block.

omerfirmak committed:
Nominator: Emit existing nominations after ballot protocol starts

In case some peers didnt make it to the balloting phase yet.

omerfirmak committed:
Nominator: Build required peers from quorum

Not the entire validator set

hewison-chris commented on source/agora/node/Validator.d:
```diff
@@ -602,22 +601,19 @@ public class Validator : FullNode, API
-        {
+        auto validators = this.getValidators();
+        quorum_conf.nodes.filter!(node_id => node_id != self_id)
+        .map!(node_id => validators[node_id].utxo)
+        .each!((utxo) {
```
```suggestion
            .map!(node_id => validators[node_id].utxo)
            .each!((utxo) {
```

hewison-chris reviewed (approved):
LGTM, just alignment.

omerfirmak commented:
Works fine in practice but it disturbed some tests.

omerfirmak committed:
Catchup: Fetch missing sigs before anything

Sigs -> Blocks -> TXs is the order that makes sense. We cant fetch
new blocks if we are missing signatures, and we cant validate new
TXs if we are missing blocks.

omerfirmak committed:
Validator: Gossip newly discovered sigs to network

omerfirmak committed:
Validator: Store early sigs and replay them after block externalization

omerfirmak commented:
so much green
